main/FatesHistoryInterfaceMod.F90

```diff
@@ -2814,6 +2815,15 @@ subroutine update_history_dyn1(this,nc,nsites,sites,bc_in)
                ! have any meaning, otherwise they are just inialization values
                notnew: if( .not.(ccohort%isnew) ) then
 
```

Comment:
As part of an upcoming refactor, we will be adding calls to daily fluxes that occur before we change plant number densities.  As already acknowledged, the number of plants present in this point of the call sequence is not the same number of plants that generated the growth flux.  This is a known issue, but it outside the scope of this PR.

Reply:
Here is the last place in the dynamics call sequence where number density has not changed yet: https://github.com/NGEET/fates/blob/sci.1.76.3_api.35.1.0/main/EDMainMod.F90#L751

Here is where we make a call to write daily nutrient fluxes, which also happens before we change number density: https://github.com/NGEET/fates/blob/sci.1.76.3_api.35.1.0/main/EDMainMod.F90#L698